pandas/core/generic.py

```diff
@@ -1110,9 +1113,19 @@ def _rename(
         else:
             # use the mapper argument
             if axis and self._get_axis_number(axis) == 1:
-                columns = mapper
+                # error: Incompatible types in assignment (expression has type
+                # "Optional[Union[Mapping[HashableT, Hashable], Callable[
+                # [HashableT], Hashable]]]", variable has type "Optional[Union[
+                # Mapping[HashableTb, Hashable], Callable[[HashableTb], Hashable
+                # ]]]")
+                columns = mapper  # type: ignore[assignment]
```

Comment:
the different TypeVars are potentially not-overlapping and therefore cannot be assigned to each other

Reply:
since it doesn't matter if we use the same TypeVar for mapper and one of either index or columns since specifying those would result in TypeError at runtime, could maybe just use two TypeVars anyway and eliminate one of these ignores?

Reply:
no longer an issue with Any